File: TP3/PYTHON/test/test_adapter.py
from src.pattern.Adapter.CoursAdapter import CoursAdapter
from src.pattern.Adapter.LegacyCours import LegacyCoursSystem

def test_adapter():
    # LegacyCours renvoie "Mathematiques;M. Durand"
    legacy = LegacyCoursSystem()
    adapter = CoursAdapter(legacy)
    cours = adapter.recuperer_cours()
    
    assert cours.nom == "Mathematiques"
    assert cours.professeur == "M. Durand"

File: TP3/JAVA/src/test/resources/TestAdapter.java
package test.resources;

import org.junit.jupiter.api.Test;

import main.resources.CoursAdapter;

import static org.junit.jupiter.api.Assertions.*;

public class TestAdapter {
    @Test
    void testAdapter() {
        Cours legacy = LegacyCoursSystem();
        CoursAdapter adapter = CoursAdapter(legacy);
        List<String> cours = adapter.getDonnees();
        
        assertEquals(cours[0], "Mathematiques");
        assertEquals(cours[1], "M. Durand");
    }
}
